src/main/java/Aly.java

```diff
@@ -0,0 +1,242 @@
+        String taskFrom = taskParts[1].trim();
+        String taskTo = taskParts[2].trim();
+        listItems[index] = new Event(taskEvent, taskFrom, taskTo);
+        index += 1;
+        System.out.println("Added this task: " + taskEvent);
+        System.out.println("You have " + Task.getTaskCounter() + " tasks in your list now.");
+        printLine();
+        return index;
+    }
+
+    private static void markTask(Task[] taskList) {
+        boolean isExit = false;
+        int index = 0;
+        index = getIndex(taskList, index);
+        while (!isExit) {
+            System.out.println("Enter 'list' to see task list, 'mark'/'unmark' with a number to toggle respective task status, '0' to exit");
+            String input = in.nextLine().trim();
+            String[] splitInput = input.split(" ");
+            String firstWord = splitInput[0];
+
+            switch (firstWord) {
+            case "list":
+                listTasks(taskList);
+                break;
+            case "mark":
+            case "unmark":
+                handleMarking(firstWord, splitInput, input, index, taskList);
+                break;
+            case "0":
+                isExit = true;
+                System.out.println("Returning to main menu!");
+                break;
+            default:
+                printLine();
+                System.out.println("Can read instructions anot? Try again!");
+                printLine();
+                break;
+            }
+        }
+    }
+
+    private static int getIndex(Task[] listItems, int index) {
+        for (int i = 0; i < 101; i++) {
+            if (listItems[i] == null) {
+                index = i;
+                break;
+            }
+        }
+        return index;
+    }
+
+    private static void handleMarking(String firstWord, String[] splitInput, String input, int index, Task[] taskList){
+            if (splitInput.length != 2) {
+                System.out.println("No task number provided!");
+            }
+            else {
+                printLine();
+                int indexNum = Integer.parseInt(splitInput[1]);
+                if (indexNum > index || indexNum <= 0) {
+                    System.out.println("Task number out of bounds!");
+                } else {
+                    if (firstWord.equals("mark")) {
+                        taskList[indexNum - 1].setDone(true);
+                        System.out.println("\"" + taskList[indexNum - 1].description + "\" marked as done!");
+                    } else {
+                        taskList[indexNum - 1].setDone(false);
+                        System.out.println("\"" + taskList[indexNum - 1].description + "\" marked as undone!");
+                    }
+                }
+            }
+            printLine();
+        }
+
+    public static void main(String[] args) {
+        printLine();
+        System.out.print("Hello! My name is \n" + LOGO);
+        initialise();
+    }
+}
```

Comment:
I suggest explaining code segments with comments. There are currently no comments, which makes it hard to understand the intended flow.